Dockerfile update

ceramicwhite opened the pull request:
- Separate deps and build stage
- Using [linuxserver/docker-baseimage-alpine](https://github.com/linuxserver/docker-baseimage-alpine) for base image of final stage
  - 30% reduced image size 
  - using s6 for pid 1
  - No more running app as root
  - Ability to add Docker Mods such as Apprise or Cloudflared via `envs`:
  ```
     environment:
        DOCKER_MODS=linuxserver/mods:universal-apprise
        DOCKER_MODS: linuxserver/mods:universal-cloudflared
        CF_ZONE_ID: zone_id
        CF_ACCOUNT_ID: acct_id
        CF_API_TOKEN: token
        CF_TUNNEL_NAME: example
        CF_TUNNEL_PASSWORD: pleasedontusethisexamplepassword
        CF_TUNNEL_CONFIG: |
          ingress:
            - hostname: test.yourdomain.url
              service: http://localhost:80
            - service: http_status:404
  ```

advplyr commented:
I don't know Docker well enough yet to understand all that is going on here.

With the `build/docker/root/etc/services.d/audiobookshelf/run` script if this is getting executed before the server is started then all of those config files won't exist yet. Those config files are created in `/server/Db.js`. Is it possible that you already had these config files created before v2.2.9?
The ownership of those config files was fixed in 2.2.9 https://github.com/advplyr/audiobookshelf/issues/584

What is "abc" here `chown -R abc:abc /config/`?

ceramicwhite commented:
@advplyr Yeah those files wouldn't exist for new users, I wanted to make sure that those upgrading wouldn't have any problems also. I've cleaned it up, I was checking the folders permissions and only changing if the user `abc` was not already owner of the folder, but the base image creates `/config` already, anyways I've fixed it.
I tested the image with server files from your image pre-2.2.9 and I had no problems switching over.

The `abc` user is just the name linuxservers gives as the default unprivileged user across their fleet. I've set it so `abc` will have `UID:GID` based on what's set for `AUDIOBOOKSHELF_UID:AUDIOBOOKSHELF_GID`

Personally I would also remove the `healthcheck`, if people want it they can add it to their docker-compose.yml
